testsuite/forge/src/interface/public.rs

```diff
@@ -111,16 +145,22 @@ impl Fund for Coffer<'_> {
 pub struct PublicInfo<'t> {
     json_rpc_url: &'t str,
     chain_id: ChainId,
-
     coffer: Coffer<'t>,
+    tx_factory: TransactionFactory,
 }
 
 impl<'t> PublicInfo<'t> {
-    pub fn new(json_rpc_url: &'t str, chain_id: ChainId, coffer: Coffer<'t>) -> Self {
+    pub fn new(
+        json_rpc_url: &'t str,
+        chain_id: ChainId,
+        coffer: Coffer<'t>,
+        tx_factory: TransactionFactory,
```

Comment:
I wouldn't require a tx factory to build this type. instead we should build it internally since all you need is the chain_id which is passed in here.